### PR TITLE
GP network tests

### LINKED ISSUE #517: GP network tests
Some comments for https://www-jsdelivr-com-pr-516.onrender.com/globalping/network-tools

- Need to support urls like /globalping/network-tools/ping /globalping/network-tools/traceroute... when visited it will open the correct page and update the text in header and footer https://dl.dropboxusercontent.com/s/vkzetacffip198q/chrome_2023-03-28_21-38-00.png

- Also need to support /globalping/network-tools/ping-from-{{LOCATION}} e.g. /globalping/network-tools/ping-from-germany
- This includes, all continents, countries, cities, ISPs and ASNs
- The header text will change depending on the selected test and region. For example for europe it says "Ping from multiple locations in Europe, World" but for a city it would say "Ping from Berlin, Europe"
- Need to compile all combinations of tests and locations and put them into our sitemap.xml
- Add pages like "from-world" and "from-globe" manually for keywords

Ping in Slack for details please

### PR DESCRIPTION
**Pull request by xbpcb**
Fix #517

### CODE REVIEW

**Comment by jimaek** [comment]
- [ ] Enter should trigger the "run test" button
- [ ] The second part of location should be clickable and open it's own page.
- [ ] When changing test type using the dropdown we should refresh the page because we need to change the URL and the bottom text
- [ ] Pre-set cdn.jsdelivr.net as default endpoint same as homepage

Not sure if backend related or not:
- [ ] The second part of location should always the next largest object. e.g. if location is Germany then the second part is Europe, California -> USA, Europe -> World, China -> Asia

File: src/middleware/sitemap/index.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ let packagesPromise = updatePackages();
 
 module.exports = async (ctx) => {
 	ctx.params.page = ctx.params.page.replace(/\.xml$/, '');
-	let pages = (await readDirRecursive(viewsPath + '/pages', [ '_*' ])).map(p => path.relative(viewsPath + '/pages', p).replace(/\\/g, '/').slice(0, -5));
+	let pages = (await readDirRecursive(viewsPath + '/pages', [ '_*' ])).map(p => path.relative(viewsPath + '/pages', p).replace(/\\/g, '/').slice(0, -7));
```
Comment (MartinKolarik) [inline src/middleware/sitemap/index.js:16]: Why 7?

Comment (xbpcb) [reply src/middleware/sitemap/index.js]: You are right, i misunderstood this moment, fixed back

File: src/views/pages/globalping/network-tools.html [hunk 2]
```diff
@@ -4,7 +4,6 @@
 <link rel="ractive" href="../../components/notification.html" name="c-notification">
```
[the remark this answers is not in the document]
Comment (jimaek) [reply src/views/pages/globalping/network-tools.html:4]: It's on purpose, those pages have little value if you're already on the site.

[the remark this answers is not in the document]
Comment (MartinKolarik) [reply src/views/pages/globalping/network-tools.html:4]: Maybe if you are on the GP homepage, you don't need links to the test pages. But we need links in the opposite direction: I somehow land on https://www-jsdelivr-com-pr-557.onrender.com/globalping/network-tools/ping-from-europe and I need to be able to get to the main GP pages.

[the remark this answers is not in the document]
Comment (jimaek) [reply src/views/pages/globalping/network-tools.html:4]: Ok yeah we can add the subnav but without the dropdown

**Comment by jimaek** [comment]
@MartinKolarik if you have time please check otherwise I'll just merge it tomorrow

**Review by MartinKolarik — APPROVED** [review]
Squash when merging please.

**Pull request merged**